core/chains/evm/gas/rollups/op_l1_oracle.go

```diff
@@ -0,0 +1,440 @@
+package rollups
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils"
+
+	gethtypes "github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/smartcontractkit/chainlink/v2/common/client"
+	"github.com/smartcontractkit/chainlink/v2/common/config"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/assets"
+	evmclient "github.com/smartcontractkit/chainlink/v2/core/chains/evm/client"
+)
+
+type OptimismL1Oracle interface {
+	L1Oracle
+	GetDAGasPrice(ctx context.Context) (*big.Int, error)
```

Comment:
I think we only need this method internally so we don't have to expose this here. Which also means we don't really need this custom interface either like Arb.

Reply:
Only reason I did this is because in `op_l1_oracle_test.go` `GetDAGasPrice` is directly referenced

Reply:
Ah gotcha, ok we can keep it like this then

Reply:
Sounds good